[Blog] Release notes for upcoming June release (#678)
gitpod-io/website: src/blog/open-vsx-in-gitpod.md

---
author: JesterOrNot, svenefftinge
date: Thu June 18 2020
image: /open-vsx.jpeg
subtitle: Open-VSX in Gitpod
teaserImage: /open-vsx.jpeg
title: Search the Extension Marketplace from Your Workspace
---

> Starting today you can not only install previously downloaded VS Code extensions by dragging and dropping them into your Gitpod workspace but search the new open and vendor-neutral marketplace [Open-VSX](https://open-vsx.org).

Reviewer: Starting with "not only" + a worse way to do things is a bit confusing to me. Maybe like so:

```suggestion
Until now, you installed VS Code extensions in Gitpod by drag-and-dropping `.vsix` files into your workspace. Today, we're excited to introduce a new, simpler way: Simply search the new open-source, vendor-neutral marketplace [Open VSX](https://open-vsx.org) directly from your workspace, and install extensions in one click.
```

> ## Open-VSX Adoption

Reviewer: ```suggestion
## Open VSX Adoption
```

> Since we first announced the Open-VSX marketplace earlier this year adoption has gone really well.

Reviewer: - we've announced it once
- adoption is way faster than what we expected

```suggestion
Since we've announced the Open VSX marketplace a few weeks ago, adoption has been extremely quick.
```

> To seed the registry, we started out publishing the most important extensions ourselves, so people could actually find the most important extensions. We could double down on adding more extensions like this, but it doesn’t seem to be necessary, because today, only a few weeks later, we already count over 200 developers publishing their extension themselves. 🎉

Reviewer: ```suggestion
To seed the registry, we've initially added the most popular open-source extensions ourselves, so that early adopters could immediately find important extensions. We could double down on adding more extensions this way, but it doesn’t seem to be necessary, because today, there are already more than 200 developers publishing their own extensions to Open VSX. 🎉
```

> Adoption from the client-side is going well, too. We started out with supporting open-vsx from within [Theia](https://theia-ide.org). Just a couple of days ago, [VSCodium](https://github.com/VSCodium/vscodium#extensions-and-the-marketplace), the true open-source version of VS Code, shipped a new version that connects with Open VSX instead of the MS marketplace as well. Today, we shipped support for Open-VSX in Gitpod.

Reviewer: "Client" sounds weird (like, people who pay for Open VSX?). "Tools" sounds better.

```suggestion
Adoption from the tool-side is going well, too. We started by supporting Open VSX from within [Theia](https://theia-ide.org). Just a couple of days ago, [VSCodium](https://github.com/VSCodium/vscodium#extensions-and-the-marketplace), the true open-source version of VS Code, shipped a new release that connects directly to Open VSX instead of Microsoft's proprietary marketplace. Today, we're excited to bring you Open VSX support in Gitpod as well.
```

## Installing Extensions in Gitpod

> As you might know already, Gitpod lets you describe ready-to-code dev environments as code. That is you write configuration and code for Gitpod to prepare dev environments for your project.

Reviewer: ```suggestion
As you may already know, Gitpod lets you describe your dev environment as code. This means that you write configuration and code that makes Gitpod automatically generate ready-to-code dev environments for your project.
```

An important part of such a configuration is the set of IDE extensions you need. In Gitpod you have two choices, you can either 

> **[1] Install an extension as part of the dev environment, or**

Reviewer: ```suggestion
**[1] Install an extension for your project's dev environment, or**
```

> **[2] install an extension into your personal user profile**

Reviewer: ```suggestion
**[2] Install an extension for your personal user profile**
```

Option [1] installs the extension for everyone starting a workspace on the same project. Programming language-specific extensions or those that support certain frameworks, databases, and so on usually go here.

Some extensions however are a matter of personal taste, such as themes and keybindings. You neither want to install and manage those extensions for every project nor do you want to enforce them on your entire team. That’s when you choose option [2].

`youtube: eD_GQ2lQug8`

> ## What If I Don’t Find the Extension I’m Looking For?

Reviewer: ```suggestion
## What If I Can’t Find the Extension I’m Looking For?
```

> While more and more extensions land in open-vsx on a daily basis there still might be cases, where you are missing a certain extension that exists in the MS marketplace. There are three different things you can do.

Reviewer: ```suggestion
While more and more extensions land in Open VSX on a daily basis, there still might be cases where you are missing a certain extension that exists in the MS marketplace. There are three things you can do:
```

>  - Just download the particular extension from somewhere (e.g. GitHub) and drag and drop it into the workspace or user area.
>  - Create a PR over at https://github.com/open-vsx/publish-extensions which is the project, where we do the seeding.
>  - Ideally, you go to the extension project itself and convince the developer to publish it to open-vsx. Publishing is a one-liner, so really low effort.

Reviewer: ```suggestion
 - Ideally, you go to the extension's repository itself and kindly ask the developer to publish it to Open VSX. Publishing is a one-liner (`npx ovsx publish`), so it's really low effort.
 - Download the extension's `.vsix` file (e.g. from the extension's GitHub Releases) and drag and drop it into the IDE's Extensions view. That will install and activate it automatically.
 - Create a PR over at https://github.com/open-vsx/publish-extensions — that's where we seed the Open VSX registry by automatically publishing popular open-source extensions.
```
